### DISCUSSION TITLE
想问问pcl2自动搜索java的工作逻辑

### DISCUSSION THREAD

**Discussion by freecube0209** [Meloong-Git/PCL#5013] [提问] [answered]
想问问pcl2自动搜索java的工作逻辑。最近把电脑里的java整理到一个统一的大文件夹里，并且用的都是zip形式的openjdk包，导致pcl2无法自动搜索到。当然有有手动添加方法，但是我更愿意配置后自动搜索到。目前往环境变量path里塞java路径确实可以让java被搜索到，但是不确定这是不是最好的方法（我总不可能往path里塞七八个路径......吧?）

**Comment by MoYuan-CN** [comment]
按理来说这种情况下应当是可以正常搜索到的？
PCL 好像是会遍历目录检查是否有可用 Java 的来着

---

Java 配置部分核心代码在 [Plain Craft Launcher 2/Modules/Minecraft/ModJava.vb](https://github.com/Hex-Dragon/PCL2/blob/main/Plain%20Craft%20Launcher%202/Modules/Minecraft/ModJava.vb)

**Reply by freecube0209** [reply]
理论上是可以搜索到的。之前我用openj9的时候放在别的文件夹就是能搜索到。可能是这次我的这个java文件夹层数太深了，举个例子这是我现在的java路径D:\java\microsoft\17\mojang-java-runtime-beta\bin\javaw.exe，而以往的其他文件夹的免安装java路径是类似于D:\java\JDK\JRE\bin\javaw.exe。当然也有可能是因为之前用的java被我手动添加过？不太清楚。

**Reply by MoYuan-CN** [reply]
你重新自动搜索一边，然后传一下 PCL 日志
有空了我分析一下是没找到还是排除了

**Reply by freecube0209** [reply]
我删了环境变量path里的所有java路径，重启pcl2后，确实无法正常搜索到放置在D:\java文件夹下的7个可用java了，只有另外一个整合包自带的java还可以被搜索。以下是日志文件。
[Log1.txt](https://github.com/user-attachments/files/17529800/Log1.txt)
[Log2.txt](https://github.com/user-attachments/files/17529801/Log2.txt)
[Log3.txt](https://github.com/user-attachments/files/17529802/Log3.txt)
[Log4.txt](https://github.com/user-attachments/files/17529803/Log4.txt)
[Log5.txt](https://github.com/user-attachments/files/17529804/Log5.txt)

另外多嘴问一句，我的“高级启动选项”-“JVM参数头”一直有如下一串代码，是正常的吗？
-XX:+UseG1GC -XX:-UseAdaptiveSizePolicy -XX:-OmitStackTraceInFastThrow -Djdk.lang.Process.allowAmbiguousCommands=true -Dfml.ignoreInvalidMinecraftCertificates=True -Dfml.ignorePatchDiscrepancies=True -Dlog4j2.formatMsgNoLookups=true

**Comment by MoYuan-CN** [comment, marked as answer]
这有点奇怪…… 我看 Log2 中你刷新前缓存中还是有的，但是刷新中就不扫描那些了……
交一个 Issue ，等龙猫处理看看

---

JVM 参数头是默认的，包含了对游戏的优化以及历史上 Log4j2 出现的一个高危漏洞的修复
没有特别需要的话保持 JVM 参数默认就行 :)

**Reply by freecube0209** [reply]
好的，非常感谢
